test/Microsoft.Health.Dicom.Tests.Integration/Features/CustomTagServiceTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.CustomTag;
+using Microsoft.Health.Dicom.Core.Messages.CustomTag;
+using Microsoft.Health.Dicom.Tests.Common.Extensions;
+using Microsoft.Health.Dicom.Tests.Integration.Persistence;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Tests.Integration.Features
+{
+    public class CustomTagServiceTests : IClassFixture<CustomTagServiceTestsFixture>
+    {
+        private static readonly CancellationToken DefaultCancellationToken = new CancellationTokenSource().Token;
+
+        private readonly CustomTagServiceTestsFixture _customTagServiceTestsFixture;
+
+        public CustomTagServiceTests(CustomTagServiceTestsFixture customTagServiceTestsFixture)
+        {
+            EnsureArg.IsNotNull(customTagServiceTestsFixture, nameof(customTagServiceTestsFixture));
+            _customTagServiceTestsFixture = customTagServiceTestsFixture;
```

Comment:
I don't think we do this, but `EnsureArg` methods typically return the value I think, so they can be used like this:

```csharp
_customTagServiceTestsFixture = EnsureArg.IsNotNull(customTagServiceTestsFixture, nameof(customTagServiceTestsFixture));
```

Without this package, people typically just do:

```csharp
_customTagServiceTestsFixture = customTagServiceTestsFixture ?? throw new ArgumentNullException(nameof(customTagServiceTestsFixture));
```